### PR TITLE
gamescope: 3.12.3 -> 3.12.5

### PR DESCRIPTION
**Pull request by Scrumplex**
## Description of changes

https://github.com/ValveSoftware/gamescope/compare/3.12.3...3.12.5

## Things done

- Built on platform(s)
  - [sx] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by PedroHLC** [comment]
@Scrumplex one thing, not with the package but with the NixOS module, I have been needing to move/copy/link the Gamescope's Vulkan layer and add it to `hardware.opengl.extraPackages` so that games detect the `VkLayer_FROG_gamescope_wsi` needed for HDR.

Should we move the `share/vulkan` directory from `$out` to its own output?

**Comment by colemickens** [comment]
```
╭ zeph  ~ 7ms
╰─▶ ls /etc/profiles/per-user/cole/share/vulkan/implicit_layer.d/
╭────┬────────────────────────────────────────────────────────────────────────────────────┬──────────┬────────┬───────────────╮
│  # │                                        name                                        │   type   │  size  │   modified    │
├────┼────────────────────────────────────────────────────────────────────────────────────┼──────────┼────────┼───────────────┤
│  0 │ /etc/profiles/per-user/cole/share/vulkan/implicit_layer.d/MangoHud.x86.json        │ symlink  │  108 B │ 53 years ago  │
│  1 │ /etc/profiles/per-user/cole/share/vulkan/implicit_layer.d/MangoHud.x86_64.json     │ symlink  │  111 B │ 53 years ago  │
│  2 │ /etc/profiles/per-user/cole/share/vulkan/implicit_layer.d/libMangoApp.x86.json     │ symlink  │  111 B │ 53 years ago  │
│  3 │ /etc/profiles/per-user/cole/share/vulkan/implicit_layer.d/libMangoApp.x86_64.json  │ symlink  │  114 B │ 53 years ago  │
│  4 │ /etc/profiles/per-user/cole/share/vulkan/implicit_layer.d/vkBasalt.json            │ symlink  │  105 B │ 53 years ago  │
│  5 │ /etc/profiles/per-user/cole/share/vulkan/implicit_layer.d/vkBasalt32.json          │ symlink  │  107 B │ 53 years ago  │
╰────┴────────────────────────────────────────────────────────────────────────────────────┴──────────┴────────┴───────────────╯
```

but I still don't know exactly how it is that "vulkan"/`vulkaninfo` knows to look there...

**Comment by colemickens** [comment]
It would be nice to know the mechanism by which vulkan knows to look there, but I think it should be sufficient to have the vulkan layer symlinked into `$out/share` the way that is done for `vkbasalt`.

At least, this is the case for me, with mangohud and vkbasalt, when using home-manager->user->packages.

**Comment by colemickens** [comment]
It kind of seems like vkbasalt installs the 64 bit layer automatically, so maybe this is another build-ish issue that should be addressed upstream in gamescope. Since I don't see anything other than the binary in our `$out` right now.

**Comment by Scrumplex** [comment]
> but I still don't know exactly how it is that "vulkan"/`vulkaninfo` knows to look there...

https://github.com/KhronosGroup/Vulkan-Loader/blob/d40385b1748ae5270fcab18c5aae1a2e5a301465/docs/LoaderLayerInterface.md#linux-layer-discovery

TL;DR; `XDG_DATA_DIRS`

**Comment by Scrumplex** [comment]
I think adding another output for the vulkan layer would be okay. I guess we could put it into `lib`?

**Comment by PedroHLC** [comment]
> TL;DR; `XDG_DATA_DIRS`

Ah, thanks, so that's how. I see now that any `systemPackage` with vulkan layers gets merged into `/run/current-system/sw/share/vulkan/implicit_layer.d`.

>  I guess we could put it into lib?

LGTM. And in another PR we can discuss how to add it to `programs.gamescope`. It could be `systemPackages`, could be appending to `programs.gamescope.env`, etc.

**Review by PedroHLC — APPROVED** [review]

**Comment by PedroHLC** [comment]
Result of `nixpkgs-review pr 255293` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>bottles</li>
    <li>bottles-unwrapped</li>
    <li>gamescope</li>
    <li>gamescope.lib</li>
  </ul>
</details>
